src/app/components/fairChain/fairChain.component.ts

```diff
@@ -52,18 +64,30 @@ export class FairChainComponent implements OnInit {
   public isInEdgeEditMode() : boolean {return this.changesNode !== ChangingNode.None;}
   private stopEditMode() : void {this.changesNode = ChangingNode.None; this.changesEdge = ChangingEdge.None;}
   private makeToolIdle() : void {this.currentTool = Tools.Idle;}
+  private closeNodeRelabelPopUp() : void {
+    assert(this.isShowingRelabelPopUp, 'There is no pop up menu to close');
+    assert(this.nodeToRelableId, 'There is no node to apply the change to'); 
+    this.nodes.update({id: this.nodeToRelableId, label: this.nodeEdgeLabel});
+    this.isShowingRelabelPopUp = false;
+    this.nodeToRelableId = undefined;
```

Comment:
Are you sure that undefined is a good Idea? Maybe something like a general Id that no node would have would be better